src/Operation/First.php

```diff
@@ -24,10 +24,6 @@ final class First extends AbstractOperation
      */
     public function __invoke(): Closure
     {
-        /** @var Closure(Iterator<TKey, T>): Generator<TKey, T> $head */
-        $head = Head::of();
-
-        // Point free style.
-        return $head;
+        return Head::of();
```

Comment:
By removing this, you get 2 new minor errors in PSalm:

```
 ~/d/g/collection   feature/more-static-analysis-checks $…  ./vendor/bin/psalm --show-info=true src/Operation/First.php                                            
Scanning files...
Analyzing files...

░

INFO: MixedReturnTypeCoercion - src/Operation/First.php:23:16 - The declared return type 'Closure(Iterator<TKey:loophp\collection\Operation\First as mixed, T:loophp\collection\Operation\First as mixed>):Generator<TKey:loophp\collection\Operation\First as mixed, T:loophp\collection\Operation\First as mixed, mixed, mixed>' for loophp\collection\Operation\First::__invoke is more specific than the inferred return type 'Closure' (see https://psalm.dev/197)
     * @return Closure(Iterator<TKey, T>): Generator<TKey, T>


INFO: MixedReturnTypeCoercion - src/Operation/First.php:27:16 - The type 'Closure' is more general than the declared return type 'Closure(Iterator<TKey:loophp\collection\Operation\First as mixed, T:loophp\collection\Operation\First as mixed>):Generator<TKey:loophp\collection\Operation\First as mixed, T:loophp\collection\Operation\First as mixed, mixed, mixed>' for loophp\collection\Operation\First::__invoke (see https://psalm.dev/197)
        return Head::of();


------------------------------
No errors found!
------------------------------
2 other issues found.
------------------------------

Checks took 0.54 seconds and used 235.298MB of memory
Psalm was able to infer types for 100% of the codebase
 ~/d/g/collection   feature/more-static-analysis-checks $
```

Reply:
oooh I didn't notice those, I guess because of the `info` level. I'll see if I can fix those, if not I'll revert

Reply:
I see now why it happens, it's because `AbstractOperation::of` has the simple return type [Closure](https://github.com/loophp/collection/blob/master/src/Operation/AbstractOperation.php#L21). I think this makes sense as they are not all the same.

I don't think it should be a problem because it's just an info thing, but I've reverted this change to keep the same style as others